### PR TITLE
Add a CLI option to delete raw data upon generation of xena_matrix

### PR DESCRIPTION
**Pull request by ayan-b**
I am using this in the AWS. Basically, it deletes the raw data corresponding to a data type once the xena matrix is generated. Let me know if we need it, I will add tests and update the docs.

### CODE REVIEW

File: xena_gdc_etl/gdc2xena.py [hunk 1]
```diff
@@ -88,6 +91,11 @@ def gdc2xena(root_dir, projects, xena_dtypes):
                 dataset = GDCOmicset(project, dtype, root_dir)
             try:
                 dataset.download().transform().metadata()
+                if delete_raw_data:
+                    print("Deleting raw data ...")
+                    shutil.rmtree(
+                        os.path.join(root_dir, project, "Raw_Data", dtype)
+                    )
```
Comment (yunhailuo) [inline xena_gdc_etl/gdc2xena.py]: Hard coded string could be changed and out of sync easily. Why not using `raw_data_dir`?

Comment (ayan-b) [reply xena_gdc_etl/gdc2xena.py]: Yes, thanks!

**Comment by yunhailuo** [comment]
Code good to go. Let me know when you done testing this code on one project for all genomic data types (including methylation).

**Comment by ayan-b** [comment]
Already used this in AWS for 10+ projects. I will add unit tests and update the docs now.

**Comment by yunhailuo** [comment]
> Already used this in AWS for 10+ projects. I will add unit tests and update the docs now.

You just pushed the last commit 12 min ago yet you've tested for 10+ projects already 9 min ago (in 3 min)? Just want to make sure you were testing the exact code to be merged.

**Comment by ayan-b** [comment]
> > Already used this in AWS for 10+ projects. I will add unit tests and update the docs now.
> 
> You just pushed the last commit 12 min ago yet you've tested for 10+ projects already 9 min ago (in 3 min)? Just want to make sure you were testing the exact code to be merged.

No for 9e731520c8dc52a8668edce20741c8512d5dd822. :see_no_evil:
Will check now.

**Comment by yunhailuo** [comment]
Take your time. Let's just be simple and stupid for this. When we decide to have a test for a PR (we can say a test is not needed), the test is on the exact code to be merged (last commit) no matter whether it's an additional test or doc change.

**Comment by ayan-b** [comment]
@yunhailuo Checked in the instance.

File: README.rst [hunk 2]
```diff
@@ -99,6 +100,9 @@ Basic usage with command line tools
     mkdir -p /home/user/xena_root
     xge etl -P FM-AD -t htseq_counts htseq_fpkm htseq_fpkm-uq
 
+  If ``-D`` or ``--delete`` is specified the raw data corresponding to a datatype
+  will be deleted upon generation of the xena matrix.
+
```
Comment (yunhailuo) [inline README.rst]: Looks strange to just explain that option. For one thing, `--help` should be enough. For another thing, `--delete` is not more important than others. Explain them all in more and elegant details than `--help` does if you really want to do that.

File: tests/test_gdc2xena.py [hunk 3]
```diff
@@ -22,3 +23,21 @@ def test_unfinished_generation():
...
+
+@pytest.mark.CI
+def test_deletion_of_raw_data():
+    os.mkdir("test_deletion")
+    gdc2xena.gdc2xena(
+        root_dir="test_deletion",
+        projects=["TCGA-ACC"],
+        xena_dtypes=["muse_snv"],
+        delete_raw_data=True,
+    )
```
Comment (yunhailuo) [inline tests/test_gdc2xena.py]: Will this run the real importing for all TCGA-ACC muse_snv data? If so, too much (even for another project) for a test.

File: tests/test_gdc2xena.py [hunk 4]
```diff
@@ -22,3 +23,21 @@ def test_unfinished_generation():
...
+    path = "/test_deletion/TCGA-ACC/Raw-Data/"
+    all_files = []
+    for root, _, files in os.walk(path):
+        for file in files:
+            all_files.append(os.path.join(root, file))
+    assert all_files == []
+    shutil.rmtree("test_deletion")
```
Comment (yunhailuo) [inline tests/test_gdc2xena.py]: Should the directory Raw-Data be gone?

File: README.rst [hunk 5]
```diff
@@ -70,6 +70,7 @@ Basic usage with command line tools
     xge etl [-h] [-r ROOT]
             [-p PROJECTS [PROJECTS ...] | -P NOT_PROJECTS [NOT_PROJECTS ...]]
             [-t DATATYPE [DATATYPE ...] | -T NOT_DATATYPE [NOT_DATATYPE ...]]
+            [-D DELETE]
```
Comment (yunhailuo) [inline README.rst]: Definitely include this one.

File: tests/test_gdc2xena.py (hunk 3 above)
Comment (ayan-b) [reply tests/test_gdc2xena.py]: Its just a few KB and only 1 file.

File: tests/test_gdc2xena.py (hunk 4 above)
Comment (ayan-b) [reply tests/test_gdc2xena.py]: Its left so that next datatype does not need to create the directory once again.

File: README.rst (hunk 2 above)
Comment (ayan-b) [reply README.rst]: Removed.

File: README.rst (hunk 5 above)
Comment (ayan-b) [reply README.rst]: Kept.

File: tests/test_gdc2xena.py (hunk 4 above)
Comment (yunhailuo) [reply tests/test_gdc2xena.py]: OK

File: tests/test_gdc2xena.py (hunk 3 above)
Comment (yunhailuo) [reply tests/test_gdc2xena.py]: This file? https://portal.gdc.cancer.gov/files/9d298cc7-71d2-43f5-835d-d7ef7a43bb11

Use a smallest one: [TCGA-UVM somaticsniper_snv](https://portal.gdc.cancer.gov/repository?facetTab=files&files_sort=%5B%7B%22field%22%3A%22file_size%22%2C%22order%22%3A%22asc%22%7D%5D&filters=%7B%22op%22%3A%22and%22%2C%22content%22%3A%5B%7B%22op%22%3A%22in%22%2C%22content%22%3A%7B%22field%22%3A%22cases.project.project_id%22%2C%22value%22%3A%5B%22TCGA-UVM%22%5D%7D%7D%2C%7B%22op%22%3A%22in%22%2C%22content%22%3A%7B%22field%22%3A%22files.analysis.workflow_type%22%2C%22value%22%3A%5B%22SomaticSniper%20Variant%20Aggregation%20and%20Masking%22%5D%7D%7D%2C%7B%22op%22%3A%22in%22%2C%22content%22%3A%7B%22field%22%3A%22files.data_category%22%2C%22value%22%3A%5B%22Simple%20Nucleotide%20Variation%22%5D%7D%7D%2C%7B%22op%22%3A%22in%22%2C%22content%22%3A%7B%22field%22%3A%22files.data_format%22%2C%22value%22%3A%5B%22MAF%22%5D%7D%7D%2C%7B%22op%22%3A%22in%22%2C%22content%22%3A%7B%22field%22%3A%22files.data_type%22%2C%22value%22%3A%5B%22Masked%20Somatic%20Mutation%22%5D%7D%7D%5D%7D).

Comment (ayan-b) [reply tests/test_gdc2xena.py]: Done!

**Comment by yunhailuo** [comment]
Thanks. Looks great!

**Pull request merged**